src/main/java/screen/GameScreen.java

```diff
@@ -487,6 +495,17 @@ else if (this.enemyShipSpecialExplosionCooldown.checkFinished())
 			this.screenFinishedCooldown.reset();
 		}
 
+		// AllyShips appear when 'isAllyShipActive' is true.
+		if (itemManager.isAllyShipActive()) {
+			for (AllyShip allyShip : this.allyShips) {
+				allyShip.shoot(this.bullets);
+			}
+		}
+		// AllyShip will be removed when its cooldown ends.
+		else if (allyShips != null) {
+			allyShips.removeAll(this.allyShips);
```

Comment:
How about changing it like below? Passing itself as a parameter to `removeAll()` is not recommended for readability and structural reasons.
```suggestion
			allyShips.clear();
```